src/lib/input/input.ts

```diff
@@ -299,6 +300,14 @@ export class MatInput extends _MatInputMixinBase implements MatFormFieldControl<
     // FormsModule or ReactiveFormsModule, because Angular forms also listens to input events.
   }
 
+  /** Determines the value of the native `placeholder` attribute that should be used in the DOM. */
+  _getNativePlaceholderValue() {
```

Comment:
Add a note that we can get rid of this once we remove the legacy form-field style, since its the only style the supports promoting the placeholder to a label

Reply:
Done.